integration_tests/testsuite/test_monitoring.py

```diff
@@ -32,33 +33,56 @@ def __init__(self, url, methodName='runTest'):
 
     def runTest(self):
         payload = test_util.generate_metrics_payload()
-        _, response_code = test_util.post(self._url, payload,
-                                          test_util.METRIC_TIMEOUT)
-        self.assertEquals(response_code, 0,
-                          'Error encountered inside sample application!')
-
+        metric_name, target = payload.get('name'), payload.get('token')
         client = google.cloud.monitoring.Client()
 
-        self.assertTrue(self._read_metric(payload.get('name'),
-                                          payload.get('token'), client),
-                        'Token not found in Stackdriver monitoring!')
-
-    @retry(wait_fixed=8000, stop_max_attempt_number=10)
+        try:
+            _, response_code = test_util.post(self._url, payload,
+                                              test_util.METRIC_TIMEOUT)
+            self.assertEquals(response_code, 0,
+                              'Error encountered inside sample application!')
+
+            logging.info('trying to find {0} stored in {1}...'.format(target, metric_name))
+            start = time.time()
+            try:
+                found_inserted_token = self._read_metric(metric_name, target, client)
+                failure = None
+            except Exception as e:
+                found_inserted_token = False
+                failure = e
+            elapsed = time.time()-start
+
+            logging.info("time elapsed checking for metric: {0}s".format(elapsed))
+
+            self.assertTrue(found_inserted_token, 'Token not found in Stackdriver monitoring! Last error: {0}'.format(failure))
+        finally:
+            self._try_cleanup_metric(client, metric_name)
+
+    def _try_cleanup_metric(self, client, metric_name):
+        try:
+            descriptor = client.metric_descriptor(metric_name)
+            descriptor.delete()
+            logging.info("metric {0} deleted".format(metric_name))
+        except Exception as e:
+            logging.warning("Error when deleting metric {0}, manual cleanup might be needed: {1}"
+                            .format(metric_name, e.message))
+
+    @retry(wait_fixed=5000, stop_max_attempt_number=20)
```

Comment:
we might also want to consider using something like exponential backoff here (or another equivalently "smart" strategy)